src/main/scala/gemmini/LoadController.scala

```diff
@@ -74,40 +77,76 @@ class LoadController[T <: Data, U <: Data, V <: Data](config: GemminiArrayConfig
   val cmd_tracker = Module(new DMAReadCommandTracker(nCmds, maxBytesInMatRequest, deps_t))
 
   io.busy := cmd.valid || cmd_tracker.io.busy
-
-  // DMA IO wiring
-  io.dma.req.valid := (control_state === waiting_for_command && cmd.valid && DoLoad && cmd_tracker.io.alloc.ready) ||
-    control_state === waiting_for_dma_req_ready ||
-    (control_state === sending_rows && row_counter =/= 0.U)
-  io.dma.req.bits.vaddr := vaddr + row_counter * stride
-  io.dma.req.bits.laddr := localaddr_plus_row_counter
-  io.dma.req.bits.len := cols
-  io.dma.req.bits.repeats := Mux(stride === 0.U, rows - 1.U, 0.U)
-  io.dma.req.bits.scale := scale
-  io.dma.req.bits.has_acc_bitwidth := localaddr_plus_row_counter.is_acc_addr && !shrink
-  io.dma.req.bits.status := mstatus
+/*
+  when(io.dma_A.req.ready === true.B){
+    io.dma_A.req.valid := (control_state === waiting_for_command && cmd.valid && DoLoad && cmd_tracker.io.alloc.ready) ||
+      control_state === waiting_for_dma_req_ready ||
+      (control_state === sending_rows && row_counter =/= 0.U)
+    io.dma_A.req.bits.vaddr := vaddr + row_counter * stride
+    io.dma_A.req.bits.laddr := localaddr_plus_row_counter
+    io.dma_A.req.bits.len := cols
+    io.dma_A.req.bits.repeats := Mux(stride === 0.U, rows - 1.U, 0.U)
+    io.dma_A.req.bits.scale := scale
+    io.dma_A.req.bits.has_acc_bitwidth := localaddr_plus_row_counter.is_acc_addr && !shrink
+    io.dma_A.req.bits.status := mstatus
+  }.otherwise{
+    io.dma_B.req.valid := (control_state === waiting_for_command && cmd.valid && DoLoad && cmd_tracker.io.alloc.ready) ||
+      control_state === waiting_for_dma_req_ready ||
+      (control_state === sending_rows && row_counter =/= 0.U)
+    io.dma_B.req.bits.vaddr := vaddr + row_counter * stride
+    io.dma_B.req.bits.laddr := localaddr_plus_row_counter
+    io.dma_B.req.bits.len := cols
+    io.dma_B.req.bits.repeats := Mux(stride === 0.U, rows - 1.U, 0.U)
+    io.dma_B.req.bits.scale := scale
+    io.dma_B.req.bits.has_acc_bitwidth := localaddr_plus_row_counter.is_acc_addr && !shrink
+    io.dma_B.req.bits.status := mstatus
+  }
+ */
+  for(d <- 0 until num_dma){
+    io.dma(d).req.valid := (control_state === waiting_for_command && cmd.valid && DoLoad && cmd_tracker.io.alloc.ready) ||
```

Comment:
I think if we do it like this, then we'll make the same DMA request to multiple DMAs at once, when we should just be making it to whichever DMA is free.